vscode-client/snippets/snippets.json

```
@@ -0,0 +1,214 @@
+    },
+    ":?": {
+        "description": "[:?] expansion",
+        "prefix": ":?",
+        "body": [
+            "\"\\${${1:variable}:?${2:error_message}}\""
+        ]
+    },
+    ":+": {
+        "description": "[:+] expansion",
+        "prefix": ":+",
+        "body": [
+            "\"\\${${1:variable}:+${2:alternative}}\""
+        ]
+    },
+    "#": {
+        "description": "[#] expansion",
+        "prefix": "#",
+        "body": [
+            "\"\\${${1:variable}#${2:pattern}}\""
+        ]
+    },
+    "##": {
+        "description": "[##] expansion",
+        "prefix": "##",
+        "body": [
+            "\"\\${${1:variable}##${2:pattern}}\""
+        ]
+    },
+    "%": {
+        "description": "[%] expansion",
+        "prefix": "%",
+        "body": [
+            "\"\\${${1:variable}%${2:pattern}}\""
+        ]
+    },
+    "%%": {
+        "description": "[%%] expansion",
+        "prefix": "%%",
+        "body": [
+            "\"\\${${1:variable}%%${2:pattern}}\""
+        ]
+    },
+    "..": {
+        "description": "brace expansion",
+        "prefix": "..",
+        "body": [
+            "{${1:from}..${2:to}}"
+        ]
+    },
+    "echo": {
+        "description": "[echo]",
+        "prefix": "echo",
+        "body": [
+            "echo \"${1:message}\""
+        ]
+    },
+    "printf": {
+        "description": "[printf]",
+        "prefix": "printf",
+        "body": [
+            "printf '%s' \"${1:message}\""
+        ]
+    },
+    "source": {
+        "description": "[source]",
+        "prefix": "source",
+        "body": [
+            "source \"${1:path/to/file}\""
+        ]
+    },
+    "alias": {
+        "description": "[alias]",
+        "prefix": "alias",
+        "body": [
+            "alias ${1:name}=${2:value}"
+        ]
+    },
+    "cd": {
+        "description": "[cd]",
+        "prefix": "cd",
+        "body": [
+            "cd \"${1:path/to/directory}\" || echo \"${2:error_message}\" >&2"
```

This seems rather non-generic.

---

For snippets to be useful I believe they should be fairly generic and applicable for most users. A good example is the `test` snippet that is a very generic. For `cd` we assume that the user wants to handle error case like this and swallow the error (exit code would be 0 after running this). 

I would like us to remove the snippet for `cd` for now.